Create Billing template

zavierboyd opened the pull request.

Commit:
Create Billing template

zavierboyd committed:
Created security account group  cloud formation

kiwifellows commented on cloudformation/newaccount/security.iam.groups.cnf.yaml:
```diff
@@ -0,0 +1,510 @@
+            Action:
+              - sts:AssumeRole
+            Resource:
+              - !FindInMap [ RoleARN, production, adminro ]
```
Replace with - !FindInMap [RoleARN, production, adminfull]

kiwifellows commented on cloudformation/newaccount/security.iam.groups.cnf.yaml:
```diff
@@ -0,0 +1,510 @@
+            Action:
+              - sts:AssumeRole
+            Resource:
+              - !FindInMap [ RoleARN, development, adminro ]
```
Replace with - !FindInMap [ RoleARN, development, adminfull ]

kiwifellows commented on cloudformation/newaccount/security.iam.groups.cnf.yaml:
```diff
@@ -0,0 +1,510 @@
+            Action:
+              - sts:AssumeRole
+            Resource:
+              - !FindInMap [ RoleARN, billing, adminro ]
```
Replace with - !FindInMap [ RoleARN, billing, adminfull ]

kiwifellows reviewed (changes requested):
Good work @zavierboyd just found a few minor things to change and can redo PR

zavierboyd committed:
Fixed full admin groups
added quicklinks for cloudformations

kiwifellows commented on cloudformation/newaccount/security.iam.groups.cnf.yaml:
```diff
@@ -0,0 +1,510 @@
+  # Creation of IAM ARNs for organisation roles
+  RoleARN:
+    production:
+      adminro: !Join
```
Apologies I only just saw this error from running the CFN inside AWS which caused a Mapping error.
lists should be as follows for Mappings (AWS CFN is really inconsistent sometimes with lists)
```yaml
Mappings: 

  # Creation of IAM ARNs for organisation roles
  RoleARN:
    production:
      adminro:
        - !Join
          - ""
          - - "arn:aws:iam::"
            - !Ref IAMProdAccountId
            - ":role/CompanyAdminRO"
```

zavierboyd replied:
I have changed that out for using !Sub instead. Makes it more readable. I have also created a devcontainer with cfn-lint extension. Will run after each file save and check the cfn template. This will prevent mistakes and make fixing things faster.

zavierboyd committed:
fixed the securityiam groups creation
changed !Join to use !Sub. More Readable
Added a devcontainer for the cfn linter. Prevents mistakes
